charms/worker/k8s/src/charm.py

```diff
@@ -149,6 +149,26 @@ def __init__(self, *args):
             self.etcd = EtcdReactiveRequires(self)
             self.framework.observe(self.on.get_kubeconfig_action, self._get_external_kubeconfig)
 
+        # kubernetes-info
+        if self.lead_control_plane:
+            self.framework.observe(self.on.ceph_k8s_info_relation_changed, self._k8s_info)
+
+    def _k8s_info(self, event: ops.RelationChangedEvent):
+        """Send cluster information on the kubernetes-info relation.
+
+        Provide applications with cluster characteristics. This should only run on the lead
+        k8s control plane unit.
+
+        Args:
+            event: ops.RelationChangedEvent - event triggered by the relation changed hook
+        """
+        if not self.lead_control_plane:
+            return
+
+        if event.relation.name == "ceph-k8s-info":
+            rel = event.relation
```

Comment:
nit: you are not using this var anywhere else. 
`event.relation.data[self.app][...]` looks fine to me. No need for this extra assignment.

Reply:
ahh, ok, so you're letting the ceph application know where kubelet is.  Very good!  Yes, i'm feeling more and more confident we should follow the reconciler pattern here.